src/plugins/global-settings/color-schemes/index.php

```diff
@@ -0,0 +1,505 @@
+				array(
+					'type' => 'string',
+					'description' => __( 'Stackable Global Base Color Scheme', STACKABLE_I18N ),
+					'sanitize_callback' => 'sanitize_text_field',
+					'show_in_rest' => true,
+					'default' => '',
+				)
+			);
+
+			register_setting(
+				'stackable_global_settings',
+				'stackable_global_background_mode_color_scheme',
+				array(
+					'type' => 'string',
+					'description' => __( 'Stackable Global Background Mode Color Scheme', STACKABLE_I18N ),
+					'sanitize_callback' => 'sanitize_text_field',
+					'show_in_rest' => true,
+					'default' => '',
+				)
+			);
+
+			register_setting(
+				'stackable_global_settings',
+				'stackable_global_container_mode_color_scheme',
+				array(
+					'type' => 'string',
+					'description' => __( 'Stackable Global Container Mode Color Scheme', STACKABLE_I18N ),
+					'sanitize_callback' => 'sanitize_text_field',
+					'show_in_rest' => true,
+					'default' => '',
+				)
+			);
+		}
+
+		public static function sanitize_array_setting( $input ) {
+			return ! is_array( $input ) ? array( array() ) : $input;
+		}
+
+		/**
+		 * Get the Color Scheme Properties/Settings
+		 *
+		 * @param 	any $values
+		 * if $values is null, return an array. Otherwise, return an associative array
+		 * where the keys are the Color Scheme properties and the values are $values.
+		 *
+		 * Note: All keys will have the value $values.
+		 * ( E.g., if $values = array('desktop' => ''), then
+		 *  properties[ 'backgroundColor' ] = array('desktop' => '' ),
+		 *  properties[ 'headingColor' ] = array('desktop' => '' ),
+		 *  etc. )
+		 *
+		 * @return 	Array
+		 */
+		public static function get_color_scheme_properties( $values = null ) {
+			$properties = [
+				'backgroundColor',
+				'headingColor',
+				'textColor',
+				'linkColor',
+				'accentColor',
+				'buttonBackgroundColor',
+				'buttonTextColor',
+				'buttonOutlineColor'
+			];
+
+			if ( $values == null ) {
+				return $properties;
+			}
+
+			$_properties = array();
+			foreach ( $properties as $key ) {
+				$_properties[ $key ] = $values;
+			}
+
+			return $_properties;
+		}
+
+		/**-----------------------------------------------------------------------------
+		 * Global Color Scheme functions
+		 *-----------------------------------------------------------------------------*/
+		/**
+		 * Add the default global color schemes in the frontend (Base, Default Background, Default Container).
+		 * Other color schemes used by blocks will be added on `render_block` filter.
+		 *
+		 * @param String 	$current_css
+		 * @return String
+		 */
+		public function add_global_color_schemes_styles( $current_css ) {
+			$schemes_array = is_array( get_option( 'stackable_global_color_schemes' ) ) ? get_option( 'stackable_global_color_schemes' ) : [];
+
+			// Get all color schemes, including custom color schemes if any
+			$all_color_schemes = apply_filters( 'stackable_global_color_schemes.get_color_schemes', $schemes_array );
```

Comment:
For the filter, let's use `/` instead of `.` for the namespacing
```suggestion
			$all_color_schemes = apply_filters( 'stackable_global_color_schemes/get_color_schemes', $schemes_array );
```